pkg/accounts/kubernetes/kubernetes.go

```diff
@@ -63,10 +66,12 @@ type CustomKubernetesResource struct {
 
 type Account struct {
 	*account.BaseAccount
-	Name     string `json:"name,omitempty"`
-	Auth     *v1alpha2.KubernetesAuth
-	Env      Env               `json:"env,omitempty"`
-	Settings v1alpha2.FreeForm `json:"settings,omitempty"`
+	Name string `json:"name,omitempty"`
+	// if this account comes from a CRD, this field indicates its namespace
+	CrdNamespace string
```

Comment:
Namespace is available in the context. I'd rather use that as namespace is not an account setting.
```
sec = secrets.FromContextWithError(ctx) -> sec.namespace
```